src/helpers.rs

```diff
@@ -509,3 +509,40 @@ fn bytes_to_os_str<'tcx, 'a>(bytes: &'a [u8]) -> InterpResult<'tcx, &'a OsStr> {
         .map_err(|_| err_unsup_format!("{:?} is not a valid utf-8 string", bytes))?;
     Ok(&OsStr::new(s))
 }
+
+pub fn immty_from_int_checked<'tcx>(
```

Comment:
I thought a bit about this, and I think the proper way to do this is to have `ImmTy::try_from_(u)int` in librustc_mir, that returns an `Option<ImmTy>`. Then we can have `from_(u)int` just `unwrap` that, but users in Miri that care can do `.ok_or`.

Reply:
I'd be happy to do that change

Reply:
Given that these are pre-existing methods (well, one of them is), this is not a blocker. Leave a FIXME though, please.
You can even do the rustc PR in parallel with the ongoing review of this one. ;)

Reply:
Heh sure :)